gh-105563: Add a reference to DateType in datetime's documentation (#105946)
python/cpython: Doc/library/datetime.rst

   Package `dateutil <https://dateutil.readthedocs.io/en/stable/>`_
      Third-party library with expanded time zone and parsing support.

>    Repository `DateType <https://github.com/glyph/DateType>`_

Reviewer: ```suggestion
   Package `DateType <https://pypi.org/project/datetype/>`_
```

>       It can be used to improve type checking of naive vs. aware datetimes.

Reviewer: ```suggestion
      Third-party library that introduces distinct types for naive and aware datetimes.
```

>       Third-party library that introduces distinct types for naive and aware datetimes.

Reviewer 2: ```suggestion
      Third-party library that introduces distinct static types to e.g. allow static type checkers
      to differentiate between naive and aware datetimes.
```
